### DISCUSSION TITLE
translate basic foods

### DISCUSSION THREAD

**Discussion by gilbrotheraway** [mealie-recipes/mealie#4529] [Feature Request]
### First Check

- [X] I used the GitHub search to find a similar requests and didn't find it.
- [X] Checked the [tasks tagged](https://github.com/mealie-recipes/mealie/issues?q=is%3Aissue+is%3Aopen+label%3Atask+) issues and verified my feature is not covered

### Please provide a concise description of the problem that would be addressed by this feature.

I'm aware that would be huge job translating all the processed crap food American recipes might include 

### Please provide a concise description of the feature that would resolve your issue.

but please consider adding the OPTIONAL feature to import from URL with auto translation from whisper or whatever method you deem appropriate 

### Please consider and list out some caveats or tradeoffs made in your design decision

i see no reason to translate brands of foods but basic ingredients only

### Additional Information

- [ ] If this is accepted I'm willing to submit a PR to provide this feature
- [X] If this is accepted I'm willing to help maintain this feature
- [X] I'm willing to sponsor/pay a developer to do this work

**Comment by Ronaldvr** [comment]
I just installed mealie and invested a crap-ton of time to throw out all the odd and wrongly translated (and strange why was cannabis in there?) foods: Remember that it will *all* show up in the list in the UI even if one never uses it.  So what would be the use? IMO It would be better to start with a *very* basic and general *and correct* list and allow users to add at will one at a time.

**Reply by hady0815** [reply]
Hello,

I would go a little further. I would like to have the possibility to import my own file. There may be a specification for how this should look or be built. With AI, such as e.g. Open AI, you could then import and/or process the foodlist you would like.

**Comment by atiris** [comment]
I think direct translation is not suitable for these categories (foods, units and labels) depending on geographical location. If you are from a specific area, some foods do not exist there at all, some are in several other categories, many units are not used at all (and until there is a calculator between imperial and metric units, those units will be completely useless for many countries) and we also have our own units that are used more often. So are labels. A direct translation (especially if it is not connectable between individual languages) is therefore wrong in my opinion. Rather, a language data set should be created for specific country, but not by direct translation, but by creating it for the given country. I'm creating a local database that is adapted to Slovak, but I can't achieve it by editing the English names, so I can't update the translation on crowdin (not to mention the declension of units and raw materials, which need something more than a simple plural field). The translation is good for the frontend of the program, but not for the resources, tags and units. In this case, I think that the data packages for the given country would be significantly more beneficial. And during installation, the administrator chooses which packages to install for which countries. Of course, there could be some kind of linking option, but often there isn't.
